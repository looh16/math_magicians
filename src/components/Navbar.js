import React from 'react';
import { NavLink } from 'react-router-dom';
import navStyles from '../css/Navbar.module.css';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <section className={navStyles.header}>
      <div>
        <h1>Math Magicians</h1>
      </div>
      <nav className={navStyles.navBar}>
        <ul className={navStyles.menu}>
          {links.map((link) => (
            <li key={link.id}>
              <NavLink to={link.path}>
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </section>

  );
};
export default Navbar;
