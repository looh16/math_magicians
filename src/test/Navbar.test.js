/* eslint-disable no-undef */
import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Components/Navbar';

function MockHeader() {
  return (
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
}

it('Renders correctly Navbar Component', () => {
  const tree = Renderer
    .create(<MockHeader />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
