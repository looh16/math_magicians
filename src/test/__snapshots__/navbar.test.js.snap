// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Navbar Component  1`] = `
<section
  className="header"
>
  <div>
    <h1>
      Math Magicians
    </h1>
  </div>
  <nav
    className="navBar"
  >
    <ul
      className="menu"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</section>
`;
